src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -472,6 +503,31 @@ private void InitializeOutgoingReplications()
                 _log.Info("Finished initialization of outgoing replications..");
         }
 
+
+        private void UpdateDestinations()
+        {
+            Destinations = ReplicationTopology?.GetDestinations(Server.NodeTag, Database.Name).ToList();
+        }
+
+        private DatabaseRecord LoadDatabaseRecord()
+        {
+            TransactionOperationContext context;
+            using (Server.ContextPool.AllocateOperationContext(out context))
+            using (context.OpenReadTransaction())
+            {
+                var databaseRecord = Server.Cluster.ReadDatabase(context, Database.Name);
+                if (databaseRecord == null)
+                {
+                    return null;
+                }
+                if (MyDatabaseRecord == null)
+                {
+                    MyDatabaseRecord = databaseRecord;
```

Comment:
This is side effecty, and doesn't clear this field if the value is null. What is the idea here?